pipeline/dags/dag_utils/sources/monenfant.py

```diff
@@ -1,12 +1,182 @@
+            return
+
+
+TWO_MINUTES_IN_MILLISECONDS = 2 * 60 * 1000
+
+
+def search_at_location(
+    browser: Browser,
+    location: str,
+    radius: Literal[5] | Literal[10] | Literal[30] = 30,
+) -> list:
+    base_url = os.environ.get("MONENFANT_URL")
+
+    search_results = []
+
+    page = browser.new_page(base_url=base_url)
+
+    # 1. fill search form
+    page.goto("/que-recherchez-vous")
+    page.get_by_placeholder("Je cherche").click()
+    page.get_by_role("button", name="Un mode d'accueil").click()
+    page.get_by_label("Une crèche").click()
+    page.get_by_placeholder("Où ?").press_sequentially(location)
+    page.locator(".input-geoloc").locator(".spaced-items").first.click()
+
+    if radius == 10:
+        page.get_by_label("Dans un rayon autour de").press("ArrowRight")
+    elif radius == 30:
+        page.get_by_label("Dans un rayon autour de").press("ArrowRight")
+
+    # 2. submit form
+    with page.expect_response(
+        lambda response: "/search?" in response.url and response.status == 200,
+        timeout=TWO_MINUTES_IN_MILLISECONDS,
+    ) as response_info:
+        with solve_captcha(page, timeout=500):
+            page.get_by_title("Afficher la liste").click()
+
+    response = response_info.value
+    data = response.json()
+
+    total_pages = data["totalPages"]
+    nb_of_main_results = data["nbOfMainResults"]
+
+    logger.info(
+        f"{location} | {nb_of_main_results} search results | {total_pages} pages"
+    )
+
+    # 3. iterate over remaining pages
+    for _ in range(1, total_pages):
+        # go to next page
+        with page.expect_response(
+            lambda response: "/search?" in response.url and response.status == 200,
+            timeout=60000,
+        ) as response_info:
+            with solve_captcha(page, timeout=500):
+                page.click(".pagination-next > a:nth-child(1)")
+
+        response = response_info.value
+        data = response.json()
+
+        current_page = data["page"]
+        search_results += data["mainResults"]
+
+        # is this the last page ?
+        if current_page == total_pages - 1:
+            break
+
+    return search_results
+
+
+VIEW_STRUCTURE_URL = "https://monenfant.fr/web/guest/que-recherchez-vous?p_p_id=fr_monenfant_fichestructure_portlet_FicheStructurePortlet&p_p_lifecycle=2&p_p_state=normal&p_p_mode=view&p_p_resource_id=fiche-structure&p_p_cacheability=cacheLevelPage&_fr_monenfant_fichestructure_portlet_FicheStructurePortlet_cmd=get_structure_details"
+
+
+def extract_structure(structure_id: str) -> dict:
+    data = {
+        "_fr_monenfant_fichestructure_portlet_FicheStructurePortlet_)id": structure_id,
+        (
+            "_fr_monenfant_fichestructure_portlet_FicheStructurePortlet_"
+            "dureeRecherche"
+        ): "345",
+        (
+            "_fr_monenfant_fichestructure_portlet_FicheStructurePortlet_"
+            "dateDebutRecherche"
+        ): datetime.now().strftime("%d/%m/%Y"),
```

Comment:
good heavens, je sens que ce truc va pas rester d'une stabilité profonde ^^ mais rien à dire sinon.

Reply:
ouep :man_shrugging: cela dit ça n'a pas changé depuis au moins 6 mois... we'll see